CoreVisualSchema/test/unit/src/au/gov/asd/tac/constellation/graph/schema/visual/attribute/objects/BlazeNGTest.java

```diff
@@ -0,0 +1,185 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.graph.schema.visual.attribute.objects;
+
+import au.gov.asd.tac.constellation.graph.schema.visual.attribute.objects.Blaze.IllegalBlazeFormatException;
+import au.gov.asd.tac.constellation.utilities.color.ConstellationColor;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author serpens24
+ */
+public class BlazeNGTest {
+    
+    public BlazeNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of getAngle method, of class Blaze.
+     */
+    @Test
+    public void testGetAngle() {
+        System.out.println("BlazeNGTest.testGetAngle");
+        final int angle = 23;
+        Blaze instance = new Blaze(angle, ConstellationColor.YELLOW);
+        assertEquals(instance.getAngle(), angle);
+    }
+
+    /**
+     * Test of getColor method, of class Blaze.
+     */
+    @Test
+    public void testGetColor() {
+        System.out.println("BlazeNGTest.testGetColor");
+        Blaze instance = new Blaze(0, ConstellationColor.YELLOW);
+        assertEquals(instance.getColor(), ConstellationColor.YELLOW);
+    }
+
+    /**
+     * Test of valueOf method, of class Blaze.
+     */
+    @Test
+    public void testValueOf() {
+        System.out.println("BlazeNGTest.testValueOf");
+
+        assertNull(Blaze.valueOf(null));
+        assertNull(Blaze.valueOf(""));
+        assertNull(Blaze.valueOf("badsanta"));
+        
+        // invalid string (bad color)
+        try {
+            Blaze instance = Blaze.valueOf("99;fruitsalad");
+            fail("Exception not thrown");
+        } catch (IllegalBlazeFormatException ex) {
+            assertTrue("Undefined colour for blaze.".equals(ex.getMessage()));
+        } catch (Exception ex) {
+            fail("Incorrect exception thrown");
+        }
```

Comment:
wondering whether it may be better to separate out this use case into a separate unit test. The benefit there is that you can annotate the test with the expected exception (IllegalBlazeFormatException) and the expected message (Undefined colour for blaze)

Reply:
I looked at that and beleive it makes the test suite bulkier wihout any real gain, but can change if you want it that way.